src/gui/fetchserversettings.cpp

```diff
@@ -0,0 +1,125 @@
+
+
+Q_LOGGING_CATEGORY(lcfetchserversettings, "sync.fetchserversettings", QtInfoMsg)
+
+namespace {
+constexpr auto timeoutC = 20s;
+}
+
+// TODO: move to libsync?
+FetchServerSettingsJob::FetchServerSettingsJob(const OCC::AccountPtr &account, QObject *parent)
+    : QObject(parent)
+    , _account(account)
+{
+}
+
+
+void FetchServerSettingsJob::start()
+{
+    // The main flow now needs the capabilities
+    auto *job = new JsonApiJob(_account, QStringLiteral("ocs/v2.php/cloud/capabilities"), {}, {}, this);
+    job->setAuthenticationJob(isAuthJob());
+    job->setTimeout(timeoutC);
+
+    QObject::connect(job, &JsonApiJob::finishedSignal, this, [job, this] {
+        auto caps =
+            job->data().value(QStringLiteral("ocs")).toObject().value(QStringLiteral("data")).toObject().value(QStringLiteral("capabilities")).toObject();
+        qCInfo(lcfetchserversettings) << "Server capabilities" << caps;
+        if (job->ocsSuccess()) {
+            // Record that the server supports HTTP/2
+            // Actual decision if we should use HTTP/2 is done in AccessManager::createRequest
+            // TODO: http2 support is currently disabled in the client code
+            if (auto reply = job->reply()) {
+                _account->setHttp2Supported(reply->attribute(QNetworkRequest::Http2WasUsedAttribute).toBool());
+            }
+            _account->setCapabilities(caps.toVariantMap());
+            if (checkServerInfo()) {
+                auto *userJob = new JsonApiJob(_account, QStringLiteral("ocs/v2.php/cloud/user"), SimpleNetworkJob::UrlQuery{}, QNetworkRequest{}, this);
+                job->setAuthenticationJob(isAuthJob());
+                job->setTimeout(timeoutC);
+                QObject::connect(userJob, &JsonApiJob::finishedSignal, this, [userJob, this] {
+                    const auto userData = userJob->data().value(QStringLiteral("ocs")).toObject().value(QStringLiteral("data")).toObject();
+                    const QString user = userData.value(QStringLiteral("id")).toString();
+                    if (!user.isEmpty()) {
+                        _account->setDavUser(user);
+                    }
+                    const QString displayName = userData.value(QStringLiteral("display-name")).toString();
+                    if (!displayName.isEmpty()) {
+                        _account->setDavDisplayName(displayName);
+                    }
+                    runAsyncUpdates();
+                    Q_EMIT finishedSignal();
+                });
+                userJob->start();
+            }
+        }
+    });
+    job->start();
+}
+
+void FetchServerSettingsJob::runAsyncUpdates()
+{
+    // those jobs are:
+    // - never auth jobs
+    // - might get queued
+    // - have the default timeout
+    // - must not be parented by this object
+
+    // ideally we would parent them to the account, but as things are messed up by the shared pointer stuff we can't at the moment
+    // so we just set them free
```

Comment:
Maybe create an issue for this, so we don't forget about it?

Reply:
https://github.com/owncloud/client/issues/10622